lib/voucher_form/voucher_form.dart

```diff
@@ -16,6 +16,16 @@ import 'package:vouchervault/models/voucher.dart' as V;
 
 part 'voucher_form.g.dart';
 
+void _updateFieldIfEmpty<T>(
+  GlobalKey<FormBuilderState> k,
+  String field,
+  Option<T> Function() update,
+) =>
+    O
+        .fromNullable(k.currentState!.fields[field])
+        .p(O.filter((f) => f.value != null))
```

Comment:
Should be `==`. We want to update on empty values.

Might need to cover off empty strings too?